src/Storages/StorageURL.cpp

```diff
@@ -1490,23 +1513,44 @@ FormatSettings StorageURL::getFormatSettingsFromArgs(const StorageFactory::Argum
     return format_settings;
 }
 
+std::string evaluateBodySubqueryToString(ASTPtr body_ptr, const std::optional<String>& format, const ContextPtr & context)
+{
+    auto write_buffer = WriteBufferFromOwnString();
```

Comment:
We should not buffer the result in memory. Instead, it should be streamed to the remote server.